ports/mimxrt/machine_uart.c

```diff
@@ -430,7 +425,7 @@ STATIC mp_uint_t machine_uart_ioctl(mp_obj_t self_in, mp_uint_t request, mp_uint
                 ret |= MP_STREAM_POLL_RD;
             }
         }
-        if ((flags & MP_STREAM_POLL_WR)) {
+        if ((flags & MP_STREAM_POLL_WR) && (self->tx_status == kStatus_LPUART_TxIdle)) {
```

Comment:
Is it better (and possible?) to indicate a writable state when there is at least one slot available in the TX buffer?  With the change here, you have to wait until the TX buffer is completely empty before being able to send, which may slow down a stream of bytes?

Reply:
The TX buffer is internal to the NXP UART driver. I did not find an interface to the NXP driver module telling, how much space is available in the buffer. At the moment the MP code just gets a callback when the buffer is empty. But I can look for it again.
In general, it should not slow down a stream. Only the timing pattern of sending will change.

Reply:
I could chang the test for at least 1 character space.

Reply:
That's not possible. Reason: The buffered send provided by the NXP library works with a static buffer, not a ring buffer. So a new transmit request will only be accepted when the previous transfer is finished.

Reply:
OK, thanks for investigating.